gitstats/ai_providers.py

```
@@ -0,0 +1,306 @@
+
+        Raises:
+            AIProviderError: If the generation fails
+        """
+        pass
+
+    def _retry_with_backoff(self, func, *args, **kwargs):
+        """Execute a function with exponential backoff retry logic."""
+        for attempt in range(self.max_retries):
+            try:
+                return func(*args, **kwargs)
+            except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)